internal/persistence/persistence_test.go

```
@@ -776,9 +777,201 @@ func TestEncryptedWALRejectsMismatchedCrypto(t *testing.T) {
+		if !ok || string(got) != want {
+			t.Fatalf("after migration key %q = %q, want %q", key, got, want)
+		}
+	}
+	// Verify nonce sidecar was written during migration.
+	sidecarPath := nonceSidecarPath(dir, 0)
+	if _, err := os.Stat(sidecarPath); os.IsNotExist(err) {
+		t.Fatal("nonce sidecar not created during migration")
+	}
+
+	// Session 3: reopen as encrypted — data still survives.
+	s2.CloseShard(0)
```

golangci-lint (2.12.2)

[error] Error return value of `s2.CloseShard` is not checked

(errcheck)